tests/FSharp.Compiler.ComponentTests/Language/InterpolatedStringsTests.fs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.Language
+
+open System
+open Xunit
+open FSharp.Test.Compiler
+
+module InterpolatedStringsTests =
+
+    [<Fact>]
+    let ``Regression: Empty Interpolated String properly typechecks with explicit type on binding`` () =
+        Fsx """ let a:byte = $"string" """
+        |> compile
+        |> shouldFail
+        |> withSingleDiagnostic (Error 1, Line 1, Col 15, Line 1, Col 24, "This expression was expected to have type" + Environment.NewLine + "    'byte'    " + Environment.NewLine + "but here has type" + Environment.NewLine + "    'string'    ")
+
+    [<Fact>]
+    let ``Interpolated String with hole properly typechecks with explicit type on binding`` () =
```

Comment:
Could you add tests for

    let a: obj = $"string"
    let a: System.IComparable = $"string"
    let a: System.IFormattable = $"string"